crates/core/src/factories/core_fetch/test/request_queue.rs

```diff
@@ -0,0 +1,599 @@
+        agent: Some(agent_id_2.clone()),
+        url: Some(Some(Url::from_str("wss://127.0.0.1:2").unwrap())),
+        ..Default::default()
+    }
+    .build();
+    peer_store.insert(vec![agent_info_2.clone()]).await.unwrap();
+
+    let fetch = CoreFetch::new(
+        config.clone(),
+        space_id.clone(),
+        peer_store,
+        op_store,
+        mock_transport.clone(),
+    );
+
+    fetch
+        .request_ops(op_list_1.clone(), agent_id_1.clone())
+        .await
+        .unwrap();
+
+    // Wait for one request to fail, so agent is put on back off list.
+    tokio::time::timeout(Duration::from_millis(10), async {
+        loop {
+            tokio::time::sleep(Duration::from_millis(1)).await;
+            if !mock_transport
+                .requests_sent
+                .lock()
+                .unwrap()
+                .clone()
+                .is_empty()
+            {
+                break;
+            }
+        }
+    })
+    .await
+    .unwrap();
+
+    let current_number_of_requests_to_agent_1 = mock_transport
+        .requests_sent
+        .lock()
+        .unwrap()
+        .iter()
+        .filter(|(_, a)| *a != agent_url_1)
+        .count();
+
+    // Back off agent the maximum possible number of times.
+    let last_back_off_interval = {
+        let mut lock = fetch.state.lock().unwrap();
+        assert!(op_list_1.iter().all(|op_id| lock
+            .requests
+            .contains(&(op_id.clone(), agent_id_1.clone()))));
+        for _ in 0..config.num_back_off_intervals {
+            lock.back_off_list.back_off_agent(&agent_id_1);
+        }
+
+        lock.back_off_list.last_back_off_interval
+    };
+
+    // Wait for back off interval to expire. Afterwards the request should fail again and all
+    // of the agent's requests should be removed from the set.
+    tokio::time::sleep(last_back_off_interval).await;
+
+    assert!(fetch
+        .state
+        .lock()
+        .unwrap()
+        .back_off_list
+        .has_last_back_off_expired(&agent_id_1));
+
+    // Add control agent's ops to set.
+    fetch
+        .request_ops(op_list_2, agent_id_2.clone())
+        .await
+        .unwrap();
+
+    // Wait for another request attempt to agent 1, which should remove all of their requests
+    // from the set.
+    tokio::time::timeout(Duration::from_millis(10), async {
+        loop {
+            if mock_transport
+                .requests_sent
+                .lock()
+                .unwrap()
+                .iter()
+                .filter(|(_, a)| *a != agent_url_1)
+                .count()
+                > current_number_of_requests_to_agent_1
+            {
+                break;
+            }
+            tokio::time::sleep(Duration::from_millis(1)).await;
+        }
+    })
+    .await
+    .unwrap();
+
+    assert!(fetch
+        .state
+        .lock()
+        .unwrap()
+        .requests
+        .iter()
+        .all(|(_, agent_id)| *agent_id != agent_id_1),);
+}
```

Comment:
These tests don't seem to be using the mockall'd transport, is that intentional?

Reply:
done here https://github.com/holochain/kitsune2/pull/63/commits/6a3221aa87f0f42aee660813ad5588c4c942a7db